### PR TITLE
JBPM-7518 - Handle errors gracefully in the apps generation

### PR DESCRIPTION
**Pull request by tsurdilo**
@mswiderski adds exception handler to index controller. Check added as we talked about every 4 seconds if there are process errors and if so error handler shows the error.html page. 

Error page looks kinda plain now, but please tell me what you want it to look like.

screen shot 2018-10-16 at 3 27 11 pm

also changed the wait time of the "generating" modal from 3 to 5 seconds as that will cover the first check after 4 seconds for errors.

### CODE REVIEW

File: src/main/java/org/jbpm/bootstrap/service/controllers/IndexController.java [hunk 1]
```diff
@@ -165,7 +175,8 @@ public Project getProject() {
         } catch (Exception e) {
             logger.error("Error when generating project",
                          e);
-            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+            //return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```
Comment (mswiderski) [inline src/main/java/org/jbpm/bootstrap/service/controllers/IndexController.java]: remove the commented line

File: src/main/resources/templates/index.html [hunk 2]
```diff
@@ -287,7 +287,7 @@ <h4 class="card-title" th:text="#{page.deploy}"></h4>
             	setTimeout(function () {
                 	hidePleaseWait();
                 	goToEndStep();
-            	}, 3000)
+            	}, 5000)
```
Comment (mswiderski) [inline src/main/resources/templates/index.html:290]: why increasing this?

File: src/main/java/org/jbpm/bootstrap/service/controllers/IndexController.java [hunk 3]
```diff
@@ -192,24 +203,53 @@ protected String resolveApplicationType(Project project) {
         }
     }
 
-    protected void waitForGeneratedProject(File generatedProject) {
-        try {
-            long totalWaitTime = 0;
-            while (true) {
+    protected void waitForGeneratedProject(File generatedProject, long processInstanceId) throws Exception {
+        // if we had errors dont even start waiting
+        if(haveProcessInstanceErrors(processInstanceId)) {
```
Comment (mswiderski) [inline src/main/java/org/jbpm/bootstrap/service/controllers/IndexController.java]: maybe change the error check to throw directly error if found so we then have the error details already extracted?

**Review by mswiderski — COMMENTED** [review]
some comments added, UI wise look good in my opinion

File: src/main/resources/templates/index.html (hunk 2 above)
Comment (tsurdilo) [reply src/main/resources/templates/index.html:290]: as soon as this timeout is over we show the "Success" part of the page and user thinks like all is ok ..just after a second to see the error page. was not looking good to me. The 5 seconds covers the initial error check and the first 4 second check for process errors where i think majority of issues will be. wdyt? I will look into updating this part of the js soon so its more stable than having a simple timeout but for now this is best i could think of.

**Comment by tsurdilo** [comment]
@mswiderski updated - thanks yes the error message is alot better now as it comes from ExecutionError directly:

screen shot 2018-10-17 at 10 41 57 am

**Review by mswiderski — APPROVED** [review]

**Pull request merged**